nucypher/policy/conditions/offchain.py

```diff
@@ -0,0 +1,168 @@
+        )
+
+        @post_load
+        def make(self, data, **kwargs):
+            return JsonApiCondition(**data)
+
+    def __init__(
+        self,
+        endpoint: str,
+        query: Optional[str],
+        return_value_test: ReturnValueTest,
+        parameters: Optional[dict] = None,
+        condition_type: str = ConditionType.JSONAPI.value,
+    ):
+        if condition_type != self.CONDITION_TYPE:
+            raise InvalidCondition(
+                f"{self.__class__.__name__} must be instantiated with the {self.CONDITION_TYPE} type."
+            )
+
+        # validate inputs using marshmallow schema
+        data = {
+            "conditionType": condition_type,
+            "endpoint": endpoint,
+            "query": query,
+            "returnValueTest": return_value_test.as_dict(),
+        }
+        if parameters:
+            data["parameters"] = parameters
+        schema = self.Schema()
+        errors = schema.validate(data)
+        if errors:
+            raise InvalidConditionLingo(errors)
+
+        self.endpoint = endpoint
+        self.parameters = parameters
+        self.query = query
+        self.return_value_test = return_value_test
+        self.logger = self.LOGGER
+
+    def fetch(self) -> requests.Response:
+        """Fetches data from the endpoint."""
+        try:
+            response = requests.get(self.endpoint, params=self.parameters, timeout=5)
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as http_error:
+            self.logger.error(f"HTTP error occurred: {http_error}")
+            raise ConditionEvaluationFailed(
+                f"Failed to fetch endpoint {self.endpoint}: {http_error}"
+            )
+        except requests.exceptions.RequestException as request_error:
+            self.logger.error(f"Request exception occurred: {request_error}")
+            raise InvalidCondition(
+                f"Failed to fetch endpoint {self.endpoint}: {request_error}"
+            )
+
+        if response.status_code != 200:
+            self.logger.error(
+                f"Failed to fetch endpoint {self.endpoint}: {response.status_code}"
+            )
+            raise ConditionEvaluationFailed(
+                f"Failed to fetch endpoint {self.endpoint}: {response.status_code}"
+            )
+
+        return response
+
+    def deserialize_response(self, response: requests.Response) -> Any:
+        """Deserializes the JSON response from the endpoint."""
+        try:
+            data = response.json()
+        except (requests.exceptions.RequestException, ValueError) as json_error:
+            self.logger.error(f"JSON parsing error occurred: {json_error}")
+            raise ConditionEvaluationFailed(
+                f"Failed to parse JSON response: {json_error}"
+            )
+        return data
+
+    def query_response(self, data: Any) -> Any:
+        try:
+            expression = parse(self.query)
```

Comment:
Should this be done earlier? I mean, when constructing the JSONApiCondition object.

Reply:
It's already validated by the marshmallow schema to handle errors - does that cover your concern?  Parsing it into an object in the constructor implies reworking most of the tests and init-time validation.